### PR TITLE
32k Aura and Anti Weakness Crystals

### PR DESCRIPTION
**Pull request by blockparole**
I made some changes.

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java [hunk 1]
```diff
@@ -30,48 +34,128 @@
...
+                        continue;
+                    }
+                    if (checkSharpness(stack)) {
+                        newSlot = i;
+                        break;
+                    }
+                }
+
+                if (newSlot == -1) {
+                    return;
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: Should we really not attack at all when we don't have a 32k? Players probably don't want to keep switching the 32k setting on and off, so having aura function like usual (until a 32k is found) might be a better idea.

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java [hunk 2]
```diff
@@ -38,21 +39,27 @@
...
     private Setting<Boolean> autoSwitch = register(Settings.b("Auto Switch"));
     private Setting<Boolean> players = register(Settings.b("Players"));
-    private Setting<Boolean> place = register(Settings.b("Place", false));
     private Setting<Boolean> mobs = register(Settings.b("Mobs", false));
     private Setting<Boolean> animals = register(Settings.b("Animals", false));
-    private Setting<Double> range = register(Settings.d("Range", 4));
+    private Setting<Boolean> place = register(Settings.b("Place", false));
+    private Setting<Boolean> explode = register(Settings.b("Explode", false));
+    private Setting<Double>  range = register(Settings.d("Range", 4));
+    private Setting<Boolean> antiweakness = register(Settings.b("Anti Weakness", false));
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java]: Naming conventions: `antiWeakness`

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java [hunk 3]
```diff
@@ -61,9 +68,32 @@ public void onUpdate() {
...
+                            newSlot = i;
+                            break;
+                        }
+                    }
+                    // check if any swords or tools were found
+                    if (newSlot != -1) {
+                        Wrapper.getPlayer().inventory.currentItem = newSlot;
+                        switchCooldown = true;
+                    }
+                }
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java]: What's this feature for?
Weakness reduces attack damage - crystals the player hit will remain unaffected in damage. Switching to a sword (or another tool) doesn't make any sense, does it?

Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java]: Indeed with a Sword or Tool a Player can explode crystals while having the Weakness Effect.

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java (hunk 2 above)
Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java]: Gonna fix tomorrow.

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java (hunk 1 above)
Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: A lot of Players use Kami in Addition to Future for example, and use Kamis Aura for 32k Swords only, cause its attacking quite fast. I set it up this way cause i used it myself like this.
Of course i can change it.

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java (hunk 3 above)
Comment (zeroeightysix) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java]: What's keeping the player from just attacking the crystal with the item they're already holding now? Why do we switch to something else?

Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java]: While under the influence of Weakness, only with a Sword or Tool (Items with additional Attack Damage) the Player is able to break crystals, with a bare Hand or a Stack of Crystals or Obi in Hand, the attacked crystal will not break.

**Comment by zeroeightysix** [comment]
One more complaint - having auto switching off and anti weakness on doesn't mix well. Once CrystalAura automatically switches you to a tool, you stay on that tool and have to manually switch back to your crystals to place another one. It would be better to automatically switch back to the crystals after a crystal has been hit using a tool.

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java (hunk 1 above)
Comment (zeroeightysix) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: I'd rather have you change it. We can't throw people without future under the bus - and honestly, Kami should be designed with it being the 'dominant client' in mind.

Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: Fixed.

**Comment by blockparole** [comment]
> 
> 
> One more complaint - having auto switching off and anti weakness on doesn't mix well. Once CrystalAura automatically switches you to a tool, you stay on that tool and have to manually switch back to your crystals to place another one. It would be better to automatically switch back to the crystals after a crystal has been hit using a tool.

I added a flag `isAttacking` so the Slot can be saved before the Anti Weakness Switch and restored when no more attackable Crystals are found.

**Comment by zeroeightysix** [comment]
Works like a charm. Thanks!

**Pull request merged**